src/audio/kpb.c

```diff
@@ -897,6 +897,130 @@ static int kpb_reset(struct comp_dev *dev)
 	return ret;
 }
 
+#ifdef KPB_HIFI3
+static void kpb_micselect_copy(struct comp_dev *dev, struct comp_buffer __sparse_cache *sink,
+			       struct comp_buffer __sparse_cache *source, size_t size,
+			       uint32_t channels)
+{
+	struct comp_data *kpb = comp_get_drvdata(dev);
+	struct audio_stream __sparse_cache *istream = &source->stream;
+	struct audio_stream __sparse_cache *ostream = &sink->stream;
+	uint16_t ch;
+	size_t i;
+
+	buffer_stream_invalidate(source, size);
+
+	if (kpb->config.sampling_width == IPC4_DEPTH_16BIT) {
+		const ae_int16 *in_ptr = (const ae_int16 *)istream->r_ptr;
+		ae_int16x4 d16 = AE_ZERO16();
+		const size_t in_offset = kpb->num_of_in_channels * sizeof(ae_int16);
+		const size_t out_offset = kpb->mic_sel_cnt * sizeof(ae_int16);
+		const size_t samples_per_chan = size / (sizeof(uint16_t) * kpb->mic_sel_cnt);
+		ae_int16 *buf = (ae_int16 *)(ostream->addr);
+		ae_int16 *buf_end = (ae_int16 *)(ostream->end_addr);
+		ae_int16 *out_ptr = (ae_int16 *)buf;
+
+		AE_SETCBEGIN0(buf);
```

Comment:
@fkwasowi good to have another Hifi developer, will add you to other HiFi reviews.
@singalsu @andrula-song pls review.